.github/workflows/icu_exhaustive_tests.yml

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2016 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+#
+# GitHub Actions configuration for Exhaustive Tests for ICU.
+# 
+# Note: The exhaustive test configuration is in a separate file
+# so that it can be run independently from the regular builds.
+#
+# To run these tests, go to the Actions Tab of your repo on the github page, 
+# select "Exhaustive Tests for ICU" and manually trigger the workflow for your branch.
+
+name: Exhaustive Tests for ICU
+
+on:
+  push:
+    # Runs post merge on maintenance branches
+    branches:
+      - 'maint/maint*'
+    paths:
+      - 'icu4c/**'
+      - 'icu4j/**'
+      - '.github/workflows/**'
+  schedule:
+    # Runs daily on default branch (main) only.
+    # https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule
+    - cron: '0 4 * * *' # Run every day at 4:00 UTC
+  workflow_dispatch:
+    # Run manually on any branch
```

Comment:
Earlier, I missed that we ran the exhaustive tests post merge. 
I have updated this so that this runs:
1. Post merge on maintenance branches
2. Daily (cronjob) on the main branch
3. Manually on any other branch

Please review this behaviour change @mihnita @markusicu @echeran

Reply:
post-merge on maintenance branches sgtm

daily on main sounds... a bit expensive. have we had that? we have many days with no commits (sadly). maybe just weekly + manual?